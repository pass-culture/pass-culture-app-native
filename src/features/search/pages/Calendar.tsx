import React from 'react'
import { Text, View } from 'react-native'
import { CalendarList, LocaleConfig } from 'react-native-calendars'

import { Spacer } from 'ui/theme'

LocaleConfig.locales['fr'] = {
  monthNames: [
    'Janvier',
    'Février',
    'Mars',
    'Avril',
    'Mai',
    'Juin',
    'Juillet',
    'Août',
    'Septembre',
    'Octobre',
    'Novembre',
    'Décembre',
  ],
  monthNamesShort: [
    'Janv.',
    'Févr.',
    'Mars',
    'Avril',
    'Mai',
    'Juin',
    'Juil.',
    'Août',
    'Sept.',
    'Oct.',
    'Nov.',
    'Déc.',
  ],
  dayNames: ['Dimanche', 'Lundi', 'Mardi', 'Mercredi', 'Jeudi', 'Vendredi', 'Samedi'],
  dayNamesShort: ['Dim.', 'Lun.', 'Mar.', 'Mer.', 'Jeu.', 'Ven.', 'Sam.'],
}
LocaleConfig.defaultLocale = 'fr'
export const Calendar: React.FC = () => (
  <React.Fragment>
    <Spacer.TopScreen />
    <CalendarList
      minDate={'2021-02-02'}
      maxDate={'2021-12-30'}
      onDayPress={(day) => {
        console.log('selected day', day)
      }}
      monthFormat={'MMMM yyyy'}
      hideDayNames={true}
      firstDay={1}
      disableAllTouchEventsForDisabledDays={true}
      // renderHeader={(date) => <Text>Mois</Text>}
      pastScrollRange={0}
      futureScrollRange={50}
      scrollEnabled={true}
      showScrollIndicator={true}
      markedDates={{
        '2021-02-20': { disabled: true },
      }}
      renderHeader={(date) => <Text>{`${date.getMonth()} ${date.getFullYear()}`}</Text>}
      dayComponent={({ date, state, marking }) => {
        console.log(marking)
        return (
          // <View>
          //   <Text
          //     style={{
          //       textAlign: 'center',
          //       textDecorationLine: 'line-through',
          //       color: state === 'disabled' || marking['disabled'] === true ? 'gray' : 'black',
          //     }}>
          //     {date.day}
          //   </Text>
          //   <Text
          //     style={{
          //       textAlign: 'center',
          //       color: state === 'disabled' || marking['disabled'] === true ? 'gray' : 'black',
          //       fontSize: 10,
          //     }}>
          //     2€
          //   </Text>
          // </View>
          <React.Fragment>
            {state === 'disabled' || marking['disabled'] === true ? (
              <View
                style={{
                  position: 'absolute',
                  transform: [{ rotate: '135deg' }],
                  top: 8,
                  width: 28,
                  height: 1,
                  borderBottomColor: 'gray',
                  borderBottomWidth: 1,
                }}
              />
            ) : null}
            <Text>{date.day}</Text>
            <Text>19.90€</Text>
          </React.Fragment>
        )
      }}
    />
  </React.Fragment>
)
