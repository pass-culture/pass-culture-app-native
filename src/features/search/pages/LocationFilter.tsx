import { t } from '@lingui/macro'
import { useNavigation } from '@react-navigation/native'
import debounce from 'lodash.debounce'
import React, { useRef } from 'react'

import { UseNavigationType } from 'features/navigation/RootNavigator'
import { LocationType } from 'libs/algolia'
import { useGeolocation } from 'libs/geolocation'
import { _ } from 'libs/i18n'
import { Banner, BannerType } from 'ui/components/Banner'
import { PageHeader } from 'ui/components/headers/PageHeader'
import { Spacer } from 'ui/theme'

import { LocationChoice } from '../components/LocationChoice'

import { useSearch } from './SearchWrapper'

const DEBOUNCED_CALLBACK = 500

export const LocationFilter: React.FC = () => {
  const { navigate, goBack } = useNavigation<UseNavigationType>()
  const { position, requestGeolocPermission } = useGeolocation()
  const { dispatch } = useSearch()
  const debouncedGoBack = useRef(debounce(goBack, DEBOUNCED_CALLBACK)).current

  const onPressPickPlace = () => {
    if (debouncedGoBack) debouncedGoBack.cancel()
    navigate('LocationPicker')
  }

  const onPressAroundMe = () => {
    if (position === null) {
      requestGeolocPermission()
    } else {
      dispatch({
        type: 'LOCATION_AROUND_ME',
        payload: { latitude: position.latitude, longitude: position.longitude },
      })
    }
    debouncedGoBack()
  }

  const onPressEverywhere = () => {
    dispatch({ type: 'LOCATION_EVERYWHERE' })
    debouncedGoBack()
  }

  return (
    <React.Fragment>
      <Spacer.TopScreen />
      <Spacer.Column numberOfSpaces={14} />
      <Spacer.Column numberOfSpaces={6} />
      <Banner
        title={_(
          t`Seules les offres Sorties et Physiques seront affichÃ©es pour une recherche avec une localisation`
        )}
        type={BannerType.INFO}
      />
      <Spacer.Column numberOfSpaces={6} />
      <LocationChoice
        testID="pickPlace"
        locationType={LocationType.PLACE}
        arrowNext={true}
        onPress={onPressPickPlace}
      />
      <Spacer.Column numberOfSpaces={4} />
      <LocationChoice
        testID="aroundMe"
        locationType={LocationType.AROUND_ME}
        onPress={onPressAroundMe}
      />
      <Spacer.Column numberOfSpaces={4} />
      <LocationChoice
        testID="everywhere"
        locationType={LocationType.EVERYWHERE}
        onPress={onPressEverywhere}
      />
      <PageHeader title={_(t`Localisation`)} />
    </React.Fragment>
  )
}
