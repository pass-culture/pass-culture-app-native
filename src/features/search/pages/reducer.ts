import { FetchAlgoliaParameters, LocationType, AlgoliaGeolocation } from 'libs/algolia'
import { DATE_FILTER_OPTIONS } from 'libs/algolia/enums'
import { SuggestedPlace } from 'libs/place'

import { clampPrice, addOrRemove } from './reducer.helpers'

export type SearchParameters = Omit<
  FetchAlgoliaParameters,
  'hitsPerPage' | 'page' | 'sortBy' | 'keywords'
>
export type SearchState = SearchParameters & { showResults: boolean }

export const initialSearchState: SearchState = {
  aroundRadius: null,
  offerCategories: [],
  tags: [],
  offerIsDuo: false,
  offerIsFree: false,
  offerIsNew: false,
  offerTypes: {
    isDigital: false,
    isEvent: false,
    isThing: false,
  },
  beginningDatetime: null,
  endingDatetime: null,
  priceRange: null,
  locationType: LocationType.EVERYWHERE,
  geolocation: null,
  date: null,
  timeRange: null,
  showResults: false,
  place: null,
}

export const DEFAULT_TIME_RANGE = [8, 24]

export type Action =
  | { type: 'INIT' }
  | { type: 'INIT_FROM_SEE_MORE'; payload: Partial<SearchState> }
  | { type: 'PRICE_RANGE'; payload: SearchState['priceRange'] }
  | { type: 'TIME_RANGE'; payload: SearchState['timeRange'] }
  | { type: 'OFFER_TYPE'; payload: keyof SearchState['offerTypes'] }
  | { type: 'SHOW_RESULTS'; payload: boolean }
  | { type: 'SET_CATEGORY'; payload: SearchState['offerCategories'] }
  | { type: 'TOGGLE_CATEGORY'; payload: string }
  | { type: 'TOGGLE_OFFER_FREE' }
  | { type: 'TOGGLE_OFFER_DUO' }
  | { type: 'TOGGLE_OFFER_NEW' }
  | { type: 'TOGGLE_DATE' }
  | { type: 'TOGGLE_HOUR' }
  | { type: 'SELECT_DATE_FILTER_OPTION'; payload: DATE_FILTER_OPTIONS }
  | { type: 'SELECT_DATE'; payload: Date }
  | { type: 'LOCATION_AROUND_ME'; payload: AlgoliaGeolocation }
  | { type: 'LOCATION_EVERYWHERE' }
  | { type: 'LOCATION_PLACE'; payload: SuggestedPlace }

export const searchReducer = (state: SearchState, action: Action): SearchState => {
  switch (action.type) {
    case 'INIT':
      return initialSearchState
    case 'SHOW_RESULTS':
      return { ...state, showResults: action.payload }
    case 'INIT_FROM_SEE_MORE':
      return {
        ...state,
        ...action.payload,
        priceRange: clampPrice(action.payload.priceRange),
      }
    case 'PRICE_RANGE':
      return { ...state, priceRange: action.payload }
    case 'TIME_RANGE':
      return { ...state, timeRange: action.payload }
    case 'TOGGLE_CATEGORY':
      return {
        ...state,
        offerCategories: addOrRemove(state.offerCategories, action.payload),
      }
    case 'SET_CATEGORY':
      return { ...state, offerCategories: action.payload }
    case 'OFFER_TYPE':
      return {
        ...state,
        offerTypes: { ...state.offerTypes, [action.payload]: !state.offerTypes[action.payload] },
      }
    case 'TOGGLE_OFFER_FREE':
      return { ...state, offerIsFree: !state.offerIsFree }
    case 'TOGGLE_OFFER_DUO':
      return { ...state, offerIsDuo: !state.offerIsDuo }
    case 'TOGGLE_OFFER_NEW':
      return { ...state, offerIsNew: !state.offerIsNew }
    case 'TOGGLE_DATE':
      if (state.date) return { ...state, date: null }
      return {
        ...state,
        date: {
          option: DATE_FILTER_OPTIONS.TODAY,
          selectedDate: new Date(),
        },
      }
    case 'TOGGLE_HOUR':
      if (state.timeRange) return { ...state, timeRange: null }
      return {
        ...state,
        timeRange: [8, 24],
      }
    case 'SELECT_DATE_FILTER_OPTION':
      if (!state.date) return state
      return { ...state, date: { ...state.date, option: action.payload } }
    case 'SELECT_DATE':
      if (!state.date) return state
      return { ...state, date: { ...state.date, selectedDate: action.payload } }
    case 'LOCATION_AROUND_ME':
      return {
        ...state,
        locationType: LocationType.AROUND_ME,
        geolocation: action.payload,
        place: null,
      }
    case 'LOCATION_EVERYWHERE':
      return {
        ...state,
        locationType: LocationType.EVERYWHERE,
        geolocation: null,
        place: null,
      }
    case 'LOCATION_PLACE':
      return {
        ...state,
        locationType: LocationType.PLACE,
        geolocation: action.payload.geolocation,
        place: action.payload,
      }
    default:
      return state
  }
}
