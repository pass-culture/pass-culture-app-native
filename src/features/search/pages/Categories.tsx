import { t } from '@lingui/macro'
import { useNavigation } from '@react-navigation/native'
import debounce from 'lodash.debounce'
import React, { useRef } from 'react'
import styled from 'styled-components/native'

import { useSearch } from 'features/search/pages/SearchWrapper'
import { CATEGORY_CRITERIA } from 'libs/algolia/enums'
import { _ } from 'libs/i18n'
import { PageHeader } from 'ui/components/headers/PageHeader'
import { Validate } from 'ui/svg/icons/Validate'
import { ColorsEnum, getSpacing, Spacer, Typo } from 'ui/theme'
import { ACTIVE_OPACITY } from 'ui/theme/colors'

const ALL = 'ALL'
export const DEBOUNCED_CALLBACK = 500

export const useSelectCategory = (callback: () => void) => {
  const { searchState, dispatch } = useSearch()
  const debouncedCallback = useRef(debounce(callback, DEBOUNCED_CALLBACK)).current

  return {
    isCategorySelected: (category: string) => {
      const [selectedCategory] = [...searchState.offerCategories, ALL]
      return selectedCategory === category
    },
    selectCategory: (category: string) => () => {
      const payload = category === ALL ? [] : [category]
      dispatch({ type: 'SET_CATEGORY', payload })
      debouncedCallback()
    },
  }
}

export const Categories: React.FC = () => {
  const { goBack } = useNavigation()
  const { isCategorySelected, selectCategory } = useSelectCategory(goBack)

  return (
    <React.Fragment>
      <React.Fragment>
        <Container>
          <Spacer.TopScreen />
          <Spacer.Column numberOfSpaces={16} />

          {Object.entries(CATEGORY_CRITERIA).map(([category, { label, icon: Icon }]) => {
            const isSelected = isCategorySelected(category)
            const color2 = isSelected ? ColorsEnum.PRIMARY : ColorsEnum.SECONDARY
            const textColor = isSelected ? ColorsEnum.PRIMARY : ColorsEnum.BLACK

            return (
              <LabelContainer key={category} onPress={selectCategory(category)} testID={category}>
                <Spacer.Row numberOfSpaces={4} />
                <Icon size={getSpacing(12)} color={ColorsEnum.PRIMARY} color2={color2} />
                <Spacer.Row numberOfSpaces={2} />
                <Typo.ButtonText color={textColor}>{label}</Typo.ButtonText>
                <Spacer.Flex />
                {isSelected && <Validate color={ColorsEnum.PRIMARY} size={getSpacing(8)} />}
                <Spacer.Row numberOfSpaces={6} />
              </LabelContainer>
            )
          })}

          <Spacer.Column numberOfSpaces={30} />
        </Container>
      </React.Fragment>

      <PageHeader title={_(t`CatÃ©gories`)} />
    </React.Fragment>
  )
}

const Container = styled.ScrollView({ flex: 1 })

const LabelContainer = styled.TouchableOpacity.attrs({
  activeOpacity: ACTIVE_OPACITY,
})({
  flexDirection: 'row',
  alignItems: 'center',
  marginBottom: getSpacing(4),
})
