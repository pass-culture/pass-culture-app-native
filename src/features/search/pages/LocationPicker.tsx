import { t } from '@lingui/macro'
import { useNavigation } from '@react-navigation/native'
import debounce from 'lodash.debounce'
import React, { useEffect, useRef, useState } from 'react'
import { FlatList, TouchableOpacity } from 'react-native'
import styled from 'styled-components/native'

import { useSearch } from 'features/search/pages/SearchWrapper'
import { _ } from 'libs/i18n'
import { fetchPlaces, REGEX_STARTING_WITH_NUMBERS, SuggestedPlace } from 'libs/place'
import { PageHeader } from 'ui/components/headers/PageHeader'
import { SearchInput } from 'ui/components/inputs/SearchInput'
import { Invalidate } from 'ui/svg/icons/Invalidate'
import { getSpacing, Spacer, Typo } from 'ui/theme'
import { ACTIVE_OPACITY, ColorsEnum } from 'ui/theme/colors'

const SEARCH_DEBOUNCE_MS = 500

const RightIcon: React.FC<{ value: string; onPress: () => void }> = (props) =>
  props.value.length > 0 ? (
    <TouchableOpacity activeOpacity={ACTIVE_OPACITY} onPress={props.onPress}>
      <Invalidate size={24} />
    </TouchableOpacity>
  ) : null

export const keyExtractor = ({ geolocation, name }: SuggestedPlace) => {
  if (geolocation) return `${geolocation.latitude}-${geolocation.longitude}`
  return `${name.short}-${name.long}`
}

const PlaceHit: React.FC<{ place: SuggestedPlace; onPress: () => void }> = ({ place, onPress }) => (
  <ItemContainer onPress={onPress} testID={keyExtractor(place)}>
    <Typo.ButtonText>{place.name.short}</Typo.ButtonText>
    <Spacer.Row numberOfSpaces={1} />
    <Typo.Body>
      {REGEX_STARTING_WITH_NUMBERS.test(place.name.short)
        ? place.extraData.city
        : place.extraData.department}
    </Typo.Body>
  </ItemContainer>
)

export const LocationPicker: React.FC = () => {
  const [places, setPlaces] = useState<SuggestedPlace[]>([])
  const [value, setValue] = useState<string>('')
  const [debouncedValue, setDebouncedValue] = useState<string>(value)
  const debouncedSetValue = useRef(debounce(setDebouncedValue, SEARCH_DEBOUNCE_MS)).current

  useEffect(() => {
    if (debouncedValue.length > 0) fetchPlaces({ query: debouncedValue }).then(setPlaces)
  }, [debouncedValue])

  const resetSearch = () => {
    setValue('')
    setDebouncedValue('')
  }

  const onChangeText = (newValue: string) => {
    setValue(newValue)
    debouncedSetValue(newValue)
  }

  return (
    <React.Fragment>
      <Spacer.TopScreen />
      <Spacer.Column numberOfSpaces={14} />
      <StyledInput>
        <Spacer.Column numberOfSpaces={4} />
        <SearchInput
          value={value}
          onChangeText={onChangeText}
          placeholder={_(t`Choisir un lieu...`)}
          autoFocus={true}
          inputHeight="tall"
          RightIcon={() => <RightIcon value={value} onPress={resetSearch} />}
        />
      </StyledInput>
      <Spacer.Column numberOfSpaces={4} />
      <SuggestedPlaces places={places} />
      <PageHeader title={_(t`Choisir un lieu`)} />
    </React.Fragment>
  )
}

export const SuggestedPlaces: React.FC<{ places: SuggestedPlace[] }> = ({ places }) => {
  const { goBack } = useNavigation()
  const { dispatch } = useSearch()

  const onPickPlace = (place: SuggestedPlace) => () => {
    if (place.geolocation) {
      dispatch({ type: 'LOCATION_PLACE', payload: place })
    }
    goBack()
  }

  return (
    <FlatList
      data={places}
      keyExtractor={keyExtractor}
      renderItem={({ item: place }) => <PlaceHit place={place} onPress={onPickPlace(place)} />}
      ListEmptyComponent={React.Fragment}
      ItemSeparatorComponent={Separator}
    />
  )
}

const StyledInput = styled.View({ width: '100%', marginHorizontal: getSpacing(6) })
const ItemContainer = styled.TouchableOpacity.attrs({
  activeOpacity: ACTIVE_OPACITY,
})({
  flexDirection: 'row',
  marginHorizontal: getSpacing(6),
  paddingVertical: getSpacing(4),
  alignItems: 'center',
})
const Separator = styled.View({
  height: 2,
  backgroundColor: ColorsEnum.GREY_LIGHT,
  marginHorizontal: getSpacing(6),
})
