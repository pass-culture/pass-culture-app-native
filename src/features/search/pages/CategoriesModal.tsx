import React, { FunctionComponent } from 'react'
import { useWindowDimensions } from 'react-native'
import styled from 'styled-components/native'

import { Categories } from 'features/search/pages/Categories'
import { AppModal } from 'ui/components/modals/AppModal'
import { Close } from 'ui/svg/icons/Close'

interface Props {
  visible: boolean
  dismissModal: () => void
}

const MODAL_HEIGHT_RATIO = 3 / 4

export const CategoriesModal: FunctionComponent<Props> = ({ visible, dismissModal }) => {
  const { height: windowHeight } = useWindowDimensions()
  return (
    <AppModal
      animationOutTiming={1}
      visible={visible}
      title="Catégories"
      leftIconAccessibilityLabel={undefined}
      leftIcon={undefined}
      onLeftIconPress={undefined}
      rightIconAccessibilityLabel={'Ne pas filtrer sur les catégories et retourner aux résultats'}
      rightIcon={Close}
      onRightIconPress={dismissModal}>
      <ModalContent windowHeight={windowHeight} testID="categoriesModal">
        <Categories dismissModal={dismissModal} />
      </ModalContent>
    </AppModal>
  )
}

const ModalContent = styled.View<{ windowHeight: number }>(({ windowHeight }) => ({
  width: '100%',
  minHeight: windowHeight * MODAL_HEIGHT_RATIO,
}))
