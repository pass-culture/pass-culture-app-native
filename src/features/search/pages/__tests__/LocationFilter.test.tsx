import { fireEvent, render } from '@testing-library/react-native'
import React from 'react'

import { initialSearchState } from 'features/search/pages/reducer'
import { useGeolocation } from 'libs/geolocation'

import { LocationFilter } from '../LocationFilter'

const mockSearchState = initialSearchState
const mockDispatch = jest.fn()
const useGeolocationMock = useGeolocation as jest.Mock

jest.mock('features/search/pages/SearchWrapper', () => ({
  useSearch: () => ({
    searchState: mockSearchState,
    dispatch: mockDispatch,
  }),
}))

jest.mock('libs/geolocation', () => ({
  useGeolocation: jest.fn(() => ({
    latitude: 2,
    longitude: 40,
  })),
}))

describe('LocationFilter component', () => {
  beforeEach(() => jest.clearAllMocks())
  it('should render correctly', () => {
    const { toJSON } = render(<LocationFilter />)
    expect(toJSON()).toMatchSnapshot()
  })

  it('should dispatch actions on click (position=YES, type=AROUND_ME)', () => {
    const { getByTestId } = render(<LocationFilter />)
    fireEvent.press(getByTestId('locationChoice-aroundMe'))
    expect(mockDispatch).toHaveBeenCalledWith({
      type: 'LOCATION_AROUND_ME',
      payload: { latitude: 2, longitude: 40 },
    })
  })

  it('should not dispatch actions on click (position=NO, type=AROUND_ME)', () => {
    useGeolocationMock.mockImplementationOnce(() => null)
    const { getByTestId } = render(<LocationFilter />)
    fireEvent.press(getByTestId('locationChoice-aroundMe'))
    expect(mockDispatch).not.toHaveBeenCalled()
  })

  it('should dispatch actions on click (position=YES, type=EVERYWHERE)', () => {
    const { getByTestId } = render(<LocationFilter />)
    fireEvent.press(getByTestId('locationChoice-everywhere'))
    expect(mockDispatch).toHaveBeenCalledWith({ type: 'LOCATION_EVERYWHERE' })
  })

  it('should dispatch actions on click (position=NO, type=EVERYWHERE)', () => {
    useGeolocationMock.mockImplementationOnce(() => null)
    const { getByTestId } = render(<LocationFilter />)
    fireEvent.press(getByTestId('locationChoice-everywhere'))
    expect(mockDispatch).toHaveBeenCalledWith({ type: 'LOCATION_EVERYWHERE' })
  })
})
