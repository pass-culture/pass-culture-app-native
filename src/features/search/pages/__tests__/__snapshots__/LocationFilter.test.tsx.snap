// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationFilter component should render correctly 1`] = `
Array [
  <View
    customHeight={16}
    style={
      Array [
        Object {
          "height": 16,
        },
      ]
    }
  />,
  <RCTScrollView
    contentContainerStyle={
      Object {
        "flexGrow": 1,
        "paddingLeft": 8,
      }
    }
  >
    <View>
      <View
        numberOfSpaces={14}
        style={
          Array [
            Object {
              "height": 56,
            },
          ]
        }
      />
      <View
        numberOfSpaces={6}
        style={
          Array [
            Object {
              "height": 24,
            },
          ]
        }
      />
      <View
        style={
          Array [
            Object {
              "marginHorizontal": 24,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#f5f5f5",
                "borderRadius": 4,
                "display": "flex",
                "flexDirection": "row",
                "paddingVertical": 16,
              },
            ]
          }
        >
          <View
            numberOfSpaces={3}
            style={
              Array [
                Object {
                  "width": 12,
                },
              ]
            }
          />
          <View
            height={32}
            width={32}
          >
            <Text>
              undefined-SVG-Mock
            </Text>
          </View>
          <View
            numberOfSpaces={3}
            style={
              Array [
                Object {
                  "width": 12,
                },
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "flexShrink": 1,
                },
              ]
            }
          >
            <Text
              color="#151515"
              style={
                Array [
                  Object {
                    "color": "#151515",
                    "fontFamily": "Montserrat-SemiBold",
                    "fontSize": 12,
                    "lineHeight": 16,
                  },
                ]
              }
            >
              Seules les sorties et offres physiques seront affich√©es pour une recherche avec une localisation
            </Text>
          </View>
          <View
            numberOfSpaces={5}
            style={
              Array [
                Object {
                  "width": 20,
                },
              ]
            }
          />
        </View>
      </View>
      <View
        numberOfSpaces={6}
        style={
          Array [
            Object {
              "height": 24,
            },
          ]
        }
      />
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginHorizontal": 24,
            "opacity": 1,
          }
        }
        testID="locationChoice-pickPlace"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexBasis": 0,
                "flexDirection": "row",
                "flexGrow": 1,
                "flexShrink": 1,
              },
            ]
          }
        >
          <View
            height={48}
            width={48}
          >
            <Text>
              undefined-SVG-Mock
            </Text>
          </View>
          <View
            numberOfSpaces={2}
            style={
              Array [
                Object {
                  "width": 8,
                },
              ]
            }
          />
          <Text
            color="#151515"
            numberOfLines={3}
            style={
              Array [
                Object {
                  "color": "#151515",
                  "fontFamily": "Montserrat-Bold",
                  "fontSize": 15,
                  "lineHeight": 20,
                },
              ]
            }
          >
            Choisir un lieu
          </Text>
        </View>
        <View
          numberOfSpaces={2}
          style={
            Array [
              Object {
                "width": 8,
              },
            ]
          }
        />
        <View
          height={32}
          width={32}
        >
          <Text>
            undefined-SVG-Mock
          </Text>
        </View>
      </View>
      <View
        numberOfSpaces={4}
        style={
          Array [
            Object {
              "height": 16,
            },
          ]
        }
      />
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginHorizontal": 24,
            "opacity": 1,
          }
        }
        testID="locationChoice-aroundMe"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexBasis": 0,
                "flexDirection": "row",
                "flexGrow": 1,
                "flexShrink": 1,
              },
            ]
          }
        >
          <View
            height={48}
            width={48}
          >
            <Text>
              undefined-SVG-Mock
            </Text>
          </View>
          <View
            numberOfSpaces={2}
            style={
              Array [
                Object {
                  "width": 8,
                },
              ]
            }
          />
          <Text
            color="#151515"
            numberOfLines={3}
            style={
              Array [
                Object {
                  "color": "#151515",
                  "fontFamily": "Montserrat-Bold",
                  "fontSize": 15,
                  "lineHeight": 20,
                },
              ]
            }
          >
            Autour de moi
          </Text>
        </View>
        <View
          numberOfSpaces={10}
          style={
            Array [
              Object {
                "width": 40,
              },
            ]
          }
        />
      </View>
      <View
        numberOfSpaces={4}
        style={
          Array [
            Object {
              "height": 16,
            },
          ]
        }
      />
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginHorizontal": 24,
            "opacity": 1,
          }
        }
        testID="locationChoice-everywhere"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexBasis": 0,
                "flexDirection": "row",
                "flexGrow": 1,
                "flexShrink": 1,
              },
            ]
          }
        >
          <View
            height={48}
            width={48}
          >
            <Text>
              undefined-SVG-Mock
            </Text>
          </View>
          <View
            numberOfSpaces={2}
            style={
              Array [
                Object {
                  "width": 8,
                },
              ]
            }
          />
          <Text
            color="#eb0055"
            numberOfLines={3}
            style={
              Array [
                Object {
                  "color": "#eb0055",
                  "fontFamily": "Montserrat-Bold",
                  "fontSize": 15,
                  "lineHeight": 20,
                },
              ]
            }
          >
            Partout
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "flex-end",
                "width": 64,
              },
            ]
          }
        >
          <View
            height={32}
            testID="validateIcon"
            width={32}
          >
            <Text>
              validateIcon-SVG-Mock
            </Text>
          </View>
        </View>
        <View
          numberOfSpaces={10}
          style={
            Array [
              Object {
                "width": 40,
              },
            ]
          }
        />
      </View>
    </View>
  </RCTScrollView>,
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#eb0055",
          "position": "absolute",
          "top": 0,
          "width": "100%",
        },
      ]
    }
  >
    <View
      customHeight={16}
      style={
        Array [
          Object {
            "height": 16,
          },
        ]
      }
    />
    <View
      numberOfSpaces={2}
      style={
        Array [
          Object {
            "height": 8,
          },
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
          },
        ]
      }
    >
      <View
        positionInHeader="left"
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexBasis": 0,
              "flexDirection": "row",
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "flex-start",
              "paddingLeft": 12,
              "paddingRight": 12,
            },
          ]
        }
      >
        <View
          accessibilityLabel="Revenir en arri√®re"
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="Revenir en arri√®re"
        >
          <View
            height={32}
            testID="icon-back"
            width={32}
          >
            <Text>
              icon-back-SVG-Mock
            </Text>
          </View>
        </View>
      </View>
      <Text
        color="#ffffff"
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontFamily": "Montserrat-Regular",
              "fontSize": 15,
              "lineHeight": 20,
              "textAlign": "center",
            },
          ]
        }
      >
        Localisation
      </Text>
      <View
        positionInHeader="right"
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexBasis": 0,
              "flexDirection": "row",
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "flex-end",
              "paddingLeft": 12,
              "paddingRight": 12,
            },
          ]
        }
      />
    </View>
    <View
      numberOfSpaces={2}
      style={
        Array [
          Object {
            "height": 8,
          },
        ]
      }
    />
  </View>,
]
`;
